box-browse-sdk/src/main/java/com/box/androidsdk/browse/activities/BoxBrowseActivity.java

```diff
@@ -71,13 +76,15 @@ protected void handleBoxResponse(BoxResponse response) {
 
     protected BoxFolder getCurrentFolder() {
         Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.box_browsesdk_fragment_container);
-        if (fragment instanceof BoxBrowseFolderFragment) {
-            System.out.println("xxx ((BoxBrowseFolderFragment) fragment).getFolder() " + fragment + " x " + ((BoxBrowseFolderFragment) fragment).getFolder());
-            return ((BoxBrowseFolderFragment) fragment).getFolder();
+        BoxFolder curFolder = fragment instanceof BoxBrowseFolderFragment ?
```

Comment:
Create new folder only if we are trying to serialize it. Otherwise we can use folder with items